ksql-engine/src/main/java/io/confluent/ksql/function/udaf/max/IntegerMaxKudaf.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function.udaf.max;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.streams.kstream.Merger;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.parser.tree.Expression;
+
+public class IntegerMaxKudaf extends KsqlAggregateFunction<Integer, Integer> {
+
+  public IntegerMaxKudaf(Integer argIndexInValue) {
```

Comment:
nit: take an int rather than Integer. The former makes it clear that null is not a valid value.
nit: Use Collections.singletonList rather than Arrays.asList: it's more efficient and immutable.

Same on other classes added.